service/src/test/java/tds/assessment/services/impl/AdminSubjectServiceImplIntegrationTests.java

```diff
@@ -0,0 +1,108 @@
+package tds.assessment.services.impl;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.sql.DataSource;
+
+import java.util.HashMap;
+import java.util.Optional;
+
+import tds.assessment.SetOfAdminSubject;
+import tds.assessment.services.AdminSubjectService;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest
+@Transactional
+public class AdminSubjectServiceImplIntegrationTests {
+    @Autowired
+    private DataSource dataSource;
+
+    @Autowired
+    private AdminSubjectService adminSubjectService;
+
+    private NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Before
+    public void setUp() {
+        jdbcTemplate = new NamedParameterJdbcTemplate(dataSource);
+    }
+
+    @Test
+    public void shouldNotFindAdminSubjectByKey() {
+        Optional<SetOfAdminSubject> maybeSetOfAdminObject = adminSubjectService.findSetOfAdminByKey("(SBAC_PT)IRP-Perf-ELA-11-Summer-2015-2016");
+        assertThat(maybeSetOfAdminObject).isNotPresent();
+    }
+
+    @Test
+    public void shouldFindAdminSubjectByKey() {
+        insertRecord();
+        Optional<SetOfAdminSubject> maybeSetOfAdminObject = adminSubjectService.findSetOfAdminByKey("(SBAC_PT)IRP-Perf-ELA-11-Summer-2015-2016");
+        assertThat(maybeSetOfAdminObject.get().getKey()).isEqualTo("(SBAC_PT)IRP-Perf-ELA-11-Summer-2015-2016");
+        assertThat(maybeSetOfAdminObject.get().getSelectionAlgorithm()).isEqualTo("virtual");
+        assertThat(maybeSetOfAdminObject.get().getAssessmentId()).isEqualTo("IRP-Perf-ELA-11");
+    }
+
+    private void insertRecord() {
```

Comment:
Insert the record rather than putting inserts in the migration scripts.  Since this is the service layer we test the business logic and can leverage the @Transactional annotation